consensus/src/dag/dag_fetcher.rs

```diff
@@ -105,36 +110,41 @@ impl DagFetcher {
                     dag_reader.bitmask(local_request.node().round()),
                 )
             };
+
             let network_request = DAGMessage::from(remote_request.clone()).into_network_message();
-            if let Ok(response) = self
-                .network
-                .send_rpc_with_fallbacks(responders, network_request, Duration::from_secs(1))
-                .await
-                .and_then(DAGMessage::try_from)
-                .and_then(FetchResponse::try_from)
-                .and_then(|response| response.verify(&remote_request, &self.epoch_state.verifier))
-            {
-                let ceritified_nodes = response.certified_nodes();
-                // TODO: support chunk response or fallback to state sync
+            let mut rpc = RpcWithFallback::new(
+                responders,
+                network_request,
+                Duration::from_secs(1),
+                self.network.clone(),
+                self.time_service.clone(),
+            );
+            while let Some(response) = rpc.next().await {
+                if let Ok(response) = response
+                    .and_then(DAGMessage::try_from)
```

Comment:
not related but i remember we discussed this can be handled by the trait i.e. make the interface takes DAGMessage instead of ConsensusMessage